python/xorbits/xgboost/mars_adapters/core.py

```diff
@@ -0,0 +1,96 @@
+# Copyright 2022-2023 XProbe Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+try:
+    import xgboost
+except ImportError:  # pragma: no cover
+    xgboost = None
+
+MARS_XGBOOST_CALLABLES = {}
+
+if xgboost is not None:
+    import functools
+    import inspect
+    from collections.abc import Iterable
+    from typing import Callable, Dict, List, Optional
+
+    from ..._mars.learn.contrib.xgboost.classifier import (
+        XGBClassifier as MarsXGBClassifier,
+    )
+    from ..._mars.learn.contrib.xgboost.regressor import (
+        XGBRegressor as MarsXGBRegressor,
+    )
+    from ...core.adapter import mars_xgboost, wrap_mars_callable
+
+    class BaseXGB:
+        def __init__(self, *args, **kwargs):
+            self.mars_instance = self.mars_cls(*args, **kwargs)
+
+        def __dir__(self) -> Iterable[str]:
+            return self.mars_instance.__dir__()
+
+    class XGBClassifier(BaseXGB):
+        mars_cls = MarsXGBClassifier
+        __doc__ = MarsXGBClassifier.__doc__
```

Comment:
The docstring is not correctly attached. Check `xorbits/pandas/mars_adapters/core.py:54` for an example.